web-registry/src/lib/wallet/hooks/useAutoConnect.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from 'react';
+
+import { UseStateSetter } from 'types/react/use-state';
+
+import { AUTO_CONNECT_WALLET_KEY } from '../wallet.constants';
+import { WalletType } from '../walletsConfig/walletsConfig.types';
+import { ConnectWalletType } from './useConnectWallet';
+
+type Props = {
+  setError: UseStateSetter<unknown>;
+  setLoaded: UseStateSetter<boolean>;
+  connectWallet: ConnectWalletType;
+};
+
+export const useAutoConnect = ({
```

Comment:
could we add some comments here describing the intended purpose/use-case? i.e. it's not immediately obvious to me why we want to have an autoconnect?

Reply:
Well, it's a behavior we already had, the idea is to auto-connect the user if he already made a connection before and did not disconnect.
Will add some comments.

Reply:
thanks for the doc string there, that helps 👍